flink-end-to-end-tests/flink-rest-api-test/src/main/java/org/apache/flink/runtime/rest/tests/RestApiTestSuite.java

```diff
@@ -0,0 +1,363 @@
+	}
+
+	private static TaskManagersInfo getTaskManagers(HttpTestClient httpClient)
+		throws TimeoutException, InterruptedException, IOException {
+		httpClient.sendGetRequest("/taskmanagers", TEST_TIMEOUT);
+		HttpTestClient.SimpleHttpResponse resp = httpClient.getNextResponse();
+		Preconditions.checkState(resp.getStatus().code() == 200,
+			"Cannot fetch task manager status!");
+		return MAPPER.readValue(resp.getContent(), TaskManagersInfo.class);
+	}
+
+	private static void testMonitoringEndpointSpecs(HttpTestClient httpClient, MessageHeaders spec,
+		Map<String, String> parameterMap) {
+		try {
+			HttpMethodWrapper method = spec.getHttpMethod();
+			String path = getRestEndpointPath(spec.getTargetRestEndpointURL(), parameterMap);
+			if (spec.getRequestClass() == EmptyRequestBody.class) {
+				switch (method) {
+					case GET:
+						httpClient.sendGetRequest(path, TEST_TIMEOUT);
+						break;
+					case DELETE:
+						httpClient.sendDeleteRequest(path, TEST_TIMEOUT);
+						break;
+					default:
+						throw new UnsupportedOperationException("Cannot handle REST Test for " + path +
+						" with method " + method + ". Only GET and DELETE requests are supported!");
+				}
+
+				HttpTestClient.SimpleHttpResponse resp = httpClient.getNextResponse();
+
+				Preconditions.checkState(resp.getStatus().code() == spec.getResponseStatusCode().code(),
+					"Found mismatching status code from endpoint " + path + " with method " + method +
+					", expecting: " + spec.getResponseStatusCode().code() + ", but was: " + resp.getStatus().code());
+				// System.out.println("Found matching status for endpoint " + path + " with method  " + method);
+				@SuppressWarnings("unchecked")
+				Object responseObject = MAPPER.readValue(resp.getContent(), spec.getResponseClass());
+				Preconditions.checkNotNull(responseObject);
+				testSuccessCount += 1;
+			} else {
+				throw new UnsupportedOperationException("Cannot handle REST Test for " + path +
+					" with method " + method + ". Non-empty payload is not supported!");
+			}
+		} catch (IOException |
+			TimeoutException |
+			InterruptedException |
+			IllegalStateException |
+			NullPointerException e) {
+			testFailureCount += 1;
+			e.printStackTrace();
+		} catch (UnsupportedOperationException e) {
+			testSkipCount += 1;
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Replace target REST endpoint with actual parameters from job launch.
+	 * @param url target REST endpoint pattern from {@link MessageHeaders} specs
+	 * @param parameterMap parameter replacement map
+	 * @return actual REST URL literal
+	 */
+	private static String getRestEndpointPath(String url, Map<String, String> parameterMap)
+		throws UnsupportedOperationException {
+		for (Map.Entry<String, String> e : parameterMap.entrySet()) {
+			if (e.getValue().equals("")) {
+				if (Pattern.compile(e.getKey()).matcher(url).find()) {
+					throw new UnsupportedOperationException("No parameter replacement found for " + e.getKey());
```

Comment:
an easier way is to check for `:` in the URL. if it exists there's a parameter that wasn't replaced.